buildbot-conf/cordova.conf

```diff
@@ -0,0 +1,336 @@
+import os
+import re
+import json
+
+from buildbot.schedulers.basic import SingleBranchScheduler, AnyBranchScheduler
+from buildbot.schedulers.forcesched import ForceScheduler
+from buildbot.schedulers.timed import Nightly
+
+from buildbot.changes.gitpoller import GitPoller
+from buildbot.changes import filter as change_filter
+
+from buildbot.process.factory import BuildFactory
+from buildbot.config import BuilderConfig
+
+from buildbot.process.properties import renderer
+from buildbot.process.properties import Property as P
+from buildbot.process.properties import Interpolate as I
+
+from buildbot.steps.source.git import Git
+from buildbot.steps.transfer import FileDownload
+from buildbot.steps.shell import ShellCommand
+from buildbot.steps.master import SetProperty
+
+from buildbot.status.results import SUCCESS
+
+# config
+MEDIC_CONFIG_FILE    = os.path.join(FP, 'cordova-config.json')
+PROJECTS_CONFIG_FILE = os.path.join(FP, 'cordova-repos.json')
+
+def parse_config_file(file_name):
+    with open(file_name, 'r') as config_file:
+        return json.load(config_file)
+
+medic_config    = parse_config_file(MEDIC_CONFIG_FILE)
+projects_config = parse_config_file(PROJECTS_CONFIG_FILE)
+
+# constants
+DEFAULT_REPO_NAME = 'src'
+BASE_WORKDIR      = '.'
+TEST_APP_NAME     = 'mobilespec'
+
+OSX     = 'osx'
+LINUX   = 'linux'
+WINDOWS = 'windows'
+
+# patterns
+CORDOVA_REPO_PATTERN = r'^.*(cordova-[^\.]+)\.git$'
+
+# interpretation of every byte-sized return code as success
+ALWAYS_SUCCESS = {i: SUCCESS for i in range(0, 256)}
+
+####### UTILITIES
+
+# helper functions
+def cordova_builders():
+    return [b.name for b in c['builders'] if b.name.startswith('cordova-')]
+
+def repo_name_from_url(url):
+    match = re.match(CORDOVA_REPO_PATTERN, url)
+    if match is not None:
+        return match.group(1)
+    return DEFAULT_REPO_NAME
+
+def repo_codebase_from_name(name):
+    repo          = projects_config[name]
+    codebase_name = repo['codebase']
+    return repo['codebases'][codebase_name]
+
+def repo_url_from_name(name):
+    return repo_codebase_from_name(name)['repo']
+
+def repo_branch_from_name(name):
```

Comment:
Even if a function is short, it's a good idea to have it if it can be used in more than one place in the future. Much like a getter method on a class, it separates interface from implementation.